rofs-filtered.c

```diff
@@ -245,31 +287,39 @@ static int read_config(const char *conf_file) {
 
     while ( getline(&line, &line_size, fh) >= 0 ) {
         // Ignore comments or empty lines in the config file
-        if (! regexec(ignore_pattern, line, 0, NULL, 0)) continue;
-
-        regex = (regex_t *)malloc(sizeof(regex_t));
-        if (! regex) {
-            log_msg(LOG_ERR, "Out of memory!");
-            return -1;
-        }
+        if (! regexec(&ignore_pattern, line, 0, NULL, 0)) continue;
 
         // Remove the \n EOL
         eol = index(line, '\n');
         if (eol) *eol = '\0';
-        regcomp_res = regcomp(regex, line, REG_EXTENDED | REG_NOSUB);
+        // Test if standalone regex compiles before concatenating.
+        regcomp_res = regcomp(&pattern, line, REG_EXTENDED | REG_NOSUB);
         if ( regcomp_res ) {
-            log_regex_error(regcomp_res, regex, line);
+            log_regex_error(regcomp_res, &pattern, line);
         } else {
-            // Add regex to the stash
+            // Add regex to the buffer
             pcount++;
-            patterns = realloc(patterns, sizeof(regex_t *) * pcount);
-            patterns[pcount - 1] = regex;
+            if ( regex_append(&buffer, line, eol - line) ) {
+                log_msg(LOG_ERR, "Out of memory while scanning config.");
+                return -1;
```

Comment:
The old code was more careless about freeing resources in case of error. The current baseline tries to free everything before returning, so this will need to adapt.